pandas/core/base.py

```diff
@@ -877,6 +877,10 @@ def _values(self):
         """ the internal implementation """
         return self.values
 
+    @property
+    def empty(self):
+        return not self.values.size
```

Comment:
Could also use `self.size` instead? 
`Index.size` uses the `_values` instead of `values`. Didn't follow that recently, but I think the `values` get converted to objects for PeriodIndex, giving an unnecessary slowdown.